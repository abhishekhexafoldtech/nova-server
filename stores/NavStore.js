import { defineStore } from "pinia";
import bootstrap from "bootstrap/dist/js/bootstrap.min.js";

export const useNavStore = defineStore("NavStore", {
  // state
  state: () => {
    return {
      bootstrap: bootstrap,
      isSidenavCollapsed: false,
      showConfig: false,
      sidenavType: 'bg-white',
      navFixed: false,
      darkMode: false,
      miniNotification: false,
      breadCrumb: "Nova-Gas"
    };
  },

  actions: {
    toggleSidenavCollapsing() {
      this.isSidenavCollapsed = !this.isSidenavCollapsed;
    },
    toggleShowConfig() {
      this.showConfig = !this.showConfig;
    },
    toggleSidebarColor(payload) {
      this.sidebarColor = payload;
    },
    setSidenavType(payload) {
      this.sidenavType = payload;
    },
    toggleDarkMode() {
      this.darkMode = !this.darkMode;
    },
    toggleNavFixed() {
      this.navFixed = !this.navFixed;
    },
    setBreadCrumbs(data) {
      this.breadCrumb = data
    }
  },
});

