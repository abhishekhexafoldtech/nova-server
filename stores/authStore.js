import { defineStore } from 'pinia';
import { getToken, removeToken, setToken } from '@/utils/auth'
import {  getUserInfo, loginByWeb, logoutByWeb } from '@/api/login'

export const useAuthStore = defineStore("auth", {

  state: () => ({
    token: getToken(),
    user_info: {}, // complete user details
    page_refreshing: true,
  }),

  getters: {
    getLoginToken() {
      return this.token
    },
  },

  actions: {
	// Login
	Login(userCredInfo) {
		return new Promise((resolve, reject) => {
			// make login data set
			let data = {
				email: userCredInfo.email,
				phone: userCredInfo.phone,
				password: userCredInfo.password,
			}
			// Lets fire login with info
			loginByWeb(data).then(response => {
				if(response.status == 200) {
					let responseData = response.data
					// console.log("responseData", responseData)
					let go = true
						if (go) {
							// SITE TOKEN
							this.token = responseData.access_token
							this.user_info = responseData.data
							setToken(responseData.access_token)

							const person = { 
									"id": responseData.data.id,
									"first_name": responseData.data.first_name,
									"last_name": responseData.data.last_name,
									"email": responseData.data.email,
									"phone": responseData.data.phone,
								}
							
							// console.log('person', person)
							localStorage.setItem("user", JSON.stringify(person));

							resolve()
						}
					}
						else {
						alert('Something went wrong!')
						reject()
					}
			}).catch(error => {
				reject(error)
			})
		})
	},

	// Logout
	LogOut() {
		return new Promise((resolve, reject) => {
			this.token = ""
			removeToken()
			localStorage.clear();
			resolve()
			// logoutByWeb().then((response) => {
			// 	this.token = ""
			// 	removeToken()
			// 	resolve(response)
			// }).catch(error => {
			// 	console.log(error)
			// 	reject(error)
			// })
			// .finally(() => {
			// })
		})
	},

	// Navbar loaded or refreshed : both cases fired
	pageLoadedORRefreshed() {
		// Get user info
		return new Promise((resolve) => {
			this.page_refreshing = true;
			// Get user info
			// getUserInfo().then((response) => {
			// this.user_info = response.data.data
			// 	resolve(response);
			// })
			// Get user info from local storage
			let temp_data = localStorage.getItem("user")
			this.user_info = JSON.parse(temp_data)
		});
	},
  },
});
