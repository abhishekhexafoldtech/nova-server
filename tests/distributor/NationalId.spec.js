import { shallowMount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import { pinia } from "@/tests/mockpinia";
import { useNavStore } from "@/stores/NavStore";
import NationalId from "@/pages/distributor/add-distributor/national-id.vue";

describe("NationalId Component", () => {
  const navStore = useNavStore(pinia);
  const wrapper = shallowMount(NationalId);
  //check save button present or not
  it("renders the Save and cancel button", async () => {
    const wrapper = shallowMount(NationalId);

    // Check if the "save" button is present
    await wrapper.find(".btn-primary").trigger('click')
    expect(wrapper.find(".btn-primary").text()).toBe("Save");

    //cancel button 
    await wrapper.find(".btn-default").trigger('click')
    expect(wrapper.find(".btn-default").text()).toBe("Cancel");

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the SingleFileUpload Component", () => {
    const wrapper = shallowMount(NationalId);
     // Find the <SingleFileUpload> component within the parent component
    const singleFileUpload = wrapper.findComponent({ name: 'SingleFileUpload' });

    // Assert that the SingleFileUpload component exists within the parent component
    expect(singleFileUpload.exists()).toBe(true);
  });

  it('renders national id component', () => {
    const wrapper = shallowMount(NationalId);

    // Find the SingleFileUpload component based on its id
    const singleFileUpload = wrapper.find('#national_id');

    // Use assertions to verify that the SingleFileUpload component exists
    expect(singleFileUpload.exists()).toBe(true);
  });
});
