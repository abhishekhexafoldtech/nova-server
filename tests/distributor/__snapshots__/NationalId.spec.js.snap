// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NationalId Component > renders the Save and cancel button 1`] = `
"<div>
  <section class=\\"edit_mang_wrap\\">
    <div class=\\"mang_inner comp_inner\\">
      <div class=\\"mang_title\\">
        <h3>National ID details</h3>
      </div>
      <el-row>
        <el-col xs=\\"24\\" sm=\\"24\\" md=\\"12\\" lg=\\"6\\" xl=\\"6\\">
          <div class=\\"comp_form_box add_dist_up w355\\">
            <div class=\\"fieldrow\\">
              <p class=\\"mb-3\\">Select ID type</p>
              <el-radio-group modelvalue=\\"1\\">
                <el-radio label=\\"1\\">Ghana card</el-radio>
                <el-radio label=\\"2\\">Voter's ID</el-radio>
              </el-radio-group>
              <single-file-upload-stub iconclass=\\"iconClass\\" progress=\\"true\\" id=\\"national_id\\" value=\\"[object Object]\\"></single-file-upload-stub>
              <!--v-if-->
            </div>
          </div>
          <p class=\\"text-warning\\"> please upload national id details. </p>
        </el-col>
      </el-row>
      <div class=\\"comp_footer float-start\\">
        <nuxt-link class=\\"btn btn-default\\" to=\\"/distributor\\">Cancel</nuxt-link><button class=\\"btn btn-primary\\">Save</button>
      </div>
    </div>
  </section>
</div>
<!-- <DialogStatus
    :dialogVisible=\\"successDialog.status\\"
    :dialogTitle=\\"successDialog.title\\"
    :dialogImage=\\"successDialog.image\\"
    :dialogText=\\"successDialog.discription\\"
    :dialogTextSize=\\"successDialog.discriptionSize\\"
    :dialogTitleSize=\\"successDialog.titleSize\\"
    :leftButtonName=\\"successDialog.leftButtonName\\"
    :rightButtonName=\\"successDialog.rightButtonName\\"
    @handleLeftButton=\\"handleSendEmail\\"
    @handleRightButton=\\"handleContinueOnboarding\\"
    @dialogVisible=\\"handleCloseDialog\\"
  /> -->
<!-- with Email -->
<!-- <DialogStatus
    :dialogVisible=\\"emailDailog.status\\"
    :dialogTitle=\\"emailDailog.title\\"
    :dialogImage=\\"emailDailog.image\\"
    :dialogText=\\"emailDailog.discription\\"
    :dialogTextSize=\\"emailDailog.discriptionSize\\"
    :dialogTitleSize=\\"emailDailog.titleSize\\"
    :leftButtonName=\\"emailDailog.leftButtonName\\"
    :rightButtonName=\\"emailDailog.rightButtonName\\"
    @handleLeftButton=\\"handleCloseEmailDialog\\"
    @handleRightButton=\\"handleSendEmailWithCredentials\\"
    @dialogVisible=\\"handleCloseEmailDialog\\"
  /> -->"
`;

exports[`NationalId Component > renders the Save button 1`] = `
"<div>
  <section class=\\"edit_mang_wrap\\">
    <div class=\\"mang_inner comp_inner\\">
      <div class=\\"mang_title\\">
        <h3>National ID details</h3>
      </div>
      <el-row>
        <el-col xs=\\"24\\" sm=\\"24\\" md=\\"12\\" lg=\\"6\\" xl=\\"6\\">
          <div class=\\"comp_form_box add_dist_up w355\\">
            <div class=\\"fieldrow\\">
              <p class=\\"mb-3\\">Select ID type</p>
              <el-radio-group modelvalue=\\"1\\">
                <el-radio label=\\"1\\">Ghana card</el-radio>
                <el-radio label=\\"2\\">Voter's ID</el-radio>
              </el-radio-group>
              <single-file-upload-stub iconclass=\\"iconClass\\" progress=\\"true\\" id=\\"national_id\\" value=\\"[object Object]\\"></single-file-upload-stub>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
        </el-col>
      </el-row>
      <div class=\\"comp_footer float-start\\">
        <nuxt-link class=\\"btn btn-default\\" to=\\"/distributor\\">Cancel</nuxt-link><button class=\\"btn btn-primary\\">Save</button>
      </div>
    </div>
  </section>
</div>
<!-- <DialogStatus
    :dialogVisible=\\"successDialog.status\\"
    :dialogTitle=\\"successDialog.title\\"
    :dialogImage=\\"successDialog.image\\"
    :dialogText=\\"successDialog.discription\\"
    :dialogTextSize=\\"successDialog.discriptionSize\\"
    :dialogTitleSize=\\"successDialog.titleSize\\"
    :leftButtonName=\\"successDialog.leftButtonName\\"
    :rightButtonName=\\"successDialog.rightButtonName\\"
    @handleLeftButton=\\"handleSendEmail\\"
    @handleRightButton=\\"handleContinueOnboarding\\"
    @dialogVisible=\\"handleCloseDialog\\"
  /> -->
<!-- with Email -->
<!-- <DialogStatus
    :dialogVisible=\\"emailDailog.status\\"
    :dialogTitle=\\"emailDailog.title\\"
    :dialogImage=\\"emailDailog.image\\"
    :dialogText=\\"emailDailog.discription\\"
    :dialogTextSize=\\"emailDailog.discriptionSize\\"
    :dialogTitleSize=\\"emailDailog.titleSize\\"
    :leftButtonName=\\"emailDailog.leftButtonName\\"
    :rightButtonName=\\"emailDailog.rightButtonName\\"
    @handleLeftButton=\\"handleCloseEmailDialog\\"
    @handleRightButton=\\"handleSendEmailWithCredentials\\"
    @dialogVisible=\\"handleCloseEmailDialog\\"
  /> -->"
`;
